### PR TITLE
Unify metrics format

### PR DESCRIPTION
**Pull request by EvanLjp**

### CODE REVIEW

File: plugins/input/skywalkingv3/jvm_metric_handle.go [hunk 1]
```diff
@@ -52,82 +51,74 @@ func (h *JVMMetricHandler) Collect(ctx context.Context, req *skywalking.JVMMetri
...
+	memLabels := labels.Clone()
 	for _, mem := range metric.Memory {
 		var memType string
 		if mem.IsHeap {
 			memType = "heap"
 		} else {
 			memType = "nonheap"
 		}
+		memLabels.Replace("type", memType)
 		memCommitted := helper.NewMetricLog("skywalking_jvm_memory_committed", metric.GetTime(),
```
Comment (EvanLjp) [inline plugins/input/skywalkingv3/jvm_metric_handle.go:71]: format

File: plugins/input/skywalkingv3/jvm_metric_handle.go [hunk 2]
```diff
@@ -52,82 +51,74 @@ func (h *JVMMetricHandler) Collect(ctx context.Context, req *skywalking.JVMMetri
...
+			memLabels)
 		logs = append(logs, memCommitted)
 
-		memInit := helper.NewMetricLog("skywalking_jvm_memory_init", metric.GetTime(),
-			strconv.FormatInt(mem.Init, 10),
-			[]helper.MetricLabel{{Name: "service", Value: service}, {Name: "serviceInstance", Value: serviceInstance}, {Name: "type", Value: memType}})
+		memInit := helper.NewMetricLog("skywalking_jvm_memory_init", metric.GetTime(), float64(mem.Init), memLabels)
 		logs = append(logs, memInit)
 
 		memMax := helper.NewMetricLog("skywalking_jvm_memory_max", metric.GetTime(),
```
Comment (EvanLjp) [inline plugins/input/skywalkingv3/jvm_metric_handle.go:79]: format

File: plugins/input/skywalkingv3/jvm_metric_handle.go [hunk 3]
```diff
@@ -52,82 +51,74 @@ func (h *JVMMetricHandler) Collect(ctx context.Context, req *skywalking.JVMMetri
...
-			strconv.FormatInt(mem.Max, 10),
-			[]helper.MetricLabel{{Name: "service", Value: service}, {Name: "serviceInstance", Value: serviceInstance}, {Name: "type", Value: memType}})
+			float64(mem.Max),
+			memLabels)
 		logs = append(logs, memMax)
 
 		memUsed := helper.NewMetricLog("skywalking_jvm_memory_used", metric.GetTime(),
```
Comment (EvanLjp) [inline plugins/input/skywalkingv3/jvm_metric_handle.go:84]: format

File: plugins/input/process/input_process_cache_linux.go [hunk 4]
```diff
@@ -141,10 +141,10 @@ func (pc *processCacheLinux) GetCmdLine() string {
...
 
-func (pc *processCacheLinux) Labels(customLabels helper.KeyValues) string {
-	if len(pc.meta.labels) == 0 {
+func (pc *processCacheLinux) Labels(customLabels *helper.MetricLabels) *helper.MetricLabels {
+	if pc.meta.labels == nil {
 		if pc.stat == nil && !pc.fetchStat() {
-			return ""
+			return nil
 		}
 		processLabels := customLabels.Clone()
```
Comment (EvanLjp) [inline plugins/input/process/input_process_cache_linux.go:149]: nil clone

File: pkg/helper/log_helper.go [hunk 5]
```diff
@@ -96,25 +107,106 @@ type MetricLabel struct {
...
+		kv.keyValues[findIndex].Value = value
+	} else {
+		kv.Append(key, value)
+	}
+	kv.clearCache()
 }
 
-func (l MetricLabels) Less(i int, j int) bool {
-	return l[i].Name < l[j].Name
+func (kv *MetricLabels) Clone() *MetricLabels {
```
Comment (EvanLjp) [inline pkg/helper/log_helper.go:145]: nil clone

File: pkg/helper/log_helper.go [hunk 6]
```diff
@@ -131,42 +229,128 @@ type HistogramData struct {
...
+		if (b >= 'a' && b <= 'z') ||
+			(b >= 'A' && b <= 'Z') ||
+			(b >= '0' && b <= '9') ||
+			b == '_' {
+			continue
+		} else {
+			if newKey == nil {
+				newKey = []byte(key)
+			}
+			newKey[i] = '_'
```
Comment (yyuuttaaoo) [inline pkg/helper/log_helper.go:294]: this is the trick

**Review by yyuuttaaoo — APPROVED** [review]

**Pull request merged**